modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/monitor/Monitor.java

```diff
@@ -11,67 +11,210 @@
+            throw new RuntimeException(errorMsg);
+        }
+        DeployResource deployResource = new DeployResource();
+        BeanUtils.copyProperties(resourceEntity, deployResource);
+        DeployServiceEntity serviceEntity = findDeployServiceEntity(resourceEntity.getId());
+        CredentialDefinition credential =
+                getCredential(serviceEntity.getCsp(), serviceEntity.getUserName());
+        OrchestratorPlugin orchestratorPlugin =
+                orchestratorService.getOrchestratorPlugin(serviceEntity.getCsp());
+        ResourceMetricRequest resourceMetricRequest =
+                getResourceMetricRequest(deployResource, credential, monitorType, from,
+                        to, period);
+
+        return orchestratorPlugin.getMetricsForResource(resourceMetricRequest);
+    }
+
+
+    private DeployServiceEntity findDeployServiceEntity(UUID id) {
+        DeployServiceEntity serviceEntity =
+                deployServiceStorage.findDeployServiceById(id);
+        if (Objects.isNull(serviceEntity)) {
+            throw new EntityNotFoundException("Service not found.");
+        }
+        return serviceEntity;
+    }
+
+    private CredentialDefinition getCredential(Csp csp, String userName) {
+        CredentialDefinition credential = credentialCenter.getCredential(
+                csp, userName);
+        if (Objects.isNull(credential)) {
+            throw new EntityNotFoundException("Credential not found.");
+        }
+        return credential;
+    }
+
+
+    private ResourceMetricRequest getResourceMetricRequest(DeployResource deployResource,
+                                                           CredentialDefinition credential,
+                                                           MonitorResourceType monitorType,
+                                                           Long from, Long to, Integer period) {
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - FIVE_MINUTES_MILLISECONDS;
+        }
+        if (Objects.isNull(to)) {
+            to = System.currentTimeMillis();
+        }
+        return new ResourceMetricRequest(deployResource, credential, monitorType, from, to, period);
+    }
+
+    private ServiceMetricRequest getServiceMetricRequest(List<DeployResource> deployResources,
+                                                         CredentialDefinition credential,
+                                                         MonitorResourceType monitorType,
+                                                         Long from, Long to, Integer period) {
+        if (Objects.isNull(from)) {
+            from = System.currentTimeMillis() - FIVE_MINUTES_MILLISECONDS;
+        }
+        if (Objects.isNull(to)) {
+            to = System.currentTimeMillis();
+        }
+        return new ServiceMetricRequest(deployResources, credential, monitorType, from, to, period);
+    }
+
+    private void validateToAndFromValues(Long from, Long to) {
```

Comment:
we still a validation to check if the provided Long value is really a valid time stamp. can you add that please?

Reply:
Any Long value is a valid timestamp. How to check it? Greater the value 1672531200000 of the time '2023-01-01 00:00:00.000'? Less the Long value of nowtime?

Reply:
As discussed offline, we can have min 0 and max as LONG.MAX_VALUE

Reply:
I add the Min(0), the Max(Long.MAX_VALUE) is unnecessary. When the input value greater than Long.MAX_VALUE，the following exception will occurs firstly:  "Failed to convert value of type 'java.lang.String' to required type 'java.lang.Long'; For input string: \"3442265300000000000000000000000000\""